edgexfoundry/edgex-go
# build(snap): update edgexfoundry snap base to core20

This commit includes a number of updates to migrate the base to core20.

- postgres: updating postgres to 12.9 from 10.19. This update includes changes to the
  post-refresh hook to handle the three cases of a) new install, b) refresh from
  a postgres-10 version and c) refresh from a postgres-12 version

- snapcraft-preload: updates to use latest version and fixed cmake setup

- redis: make-install-var replaced with make-parameters

- kong-daemon: setpriv package replaced with util-linux

Signed-off-by: Siggi Skulason <siggi.skulason@canonical.com>







**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I have added unit tests for the new feature or bug fix (if not, why?) See https://github.com/canonical/edgex-checkbox-provider/pull/21
  <link to docs PR>

## Testing Instructions


Test by building the edgexfoundry snap and running a) checkbox tests and b) TAF tests to confirm that there are no regressions.

Test the refresh by doing
1: new install
```
sudo snap remove --purge edgexfoundry
sudo snap install ./edgexfoundry_2.2.0-dev.13_amd64.snap --dangerous
```
2: refresh from Ireland
```
sudo snap remove --purge edgexfoundry
sudo snap install edgexfoundry --channel=2.0/stable
sudo snap install ./edgexfoundry_2.2.0-dev.13_amd64.snap --dangerous
```
Expected result: This will fail, as the epoch has been updated to 6. We will provide a later build with a 6* epoch, which will contain the pre-refresh hook code required to upgrade.


3: refresh from current build
```
sudo snap remove --purge edgexfoundry
sudo snap install ./edgexfoundry_2.2.0-dev.13_amd64.snap --dangerous
sudo snap install ./edgexfoundry_2.2.0-dev.13_amd64.snap --dangerous
```





## New Dependency Instructions (If applicable)
